python/beeai_framework/errors.py

```diff
@@ -14,17 +14,20 @@
 
 from asyncio import CancelledError
 from collections.abc import Generator
+from typing import Any
 
 
 def _format_error_message(e: BaseException, *, offset: int = 0, strip_traceback: bool = True) -> str:
     cls = type(e).__name__
     module = type(e).__module__
+    context_str = ""
+    if isinstance(e, FrameworkError) and e.context:
+        context_str = ", ".join([f"{k}={v}" for k, v in e.context.items()])
 
     prefix = "  " * offset
-    formatted = f"{cls}({module}): {e!s}"
+    formatted = f"{cls}({module}): {e!s}{f' [{context_str}]' if context_str else ''}"
```

Comment:
I would argue that the context can be sometimes quite large. Therefore I would change the formatting to be like.

```txt
formatted = f"{cls}({module}): {e!s}"
if context_str:
   formatted += "\n{prefix}Context: {to_json(context_str)}"
```

Reply:
Before this change:
```
FrameworkError(beeai_framework.errors): Function 'getUser' has failed. [workflow=activity_planner, provider=ollama, chat_model=granite3.2:8b]
  FrameworkError(beeai_framework.errors): Cannot retrieve data from the API.
    ValueError(builtins): User with Given ID Does not exist!
```
With this kind of change we'd then have 
```
FrameworkError(beeai_framework.errors): Function 'getUser' has failed.
Context: {"chat_model": "granite3.2:8b", "provider": "ollama", "workflow": "activity_planner"}
  FrameworkError(beeai_framework.errors): Cannot retrieve data from the API.
    ValueError(builtins): User with Given ID Does not exist!

```